Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -352,69 +505,102 @@ def get_random_redirection(original: Transition, all_redirections: Iterable[Tran
 
 def set_transitions_map():  # noqa: PLR0915 # TODO: Break up in smaller functions
     transitions_map.clear()
-    initialize_connections_left()
     remove_disabled_exits()
+    initialize_connections_left()
+
     if not CONFIGS.SKIP_JAGUAR:
-        starting_default = TRANSITION_INFOS_DICT_RANDO[starting_area].default_entrance
+        starting_default = transition_infos_dict_rando[starting_area].default_entrance
         tutorial_original = Transition(from_=LevelCRC.JAGUAR, to=LevelCRC.PLANE_CUTSCENE)
         tutorial_redirect = Transition(from_=starting_default, to=starting_area)
         transitions_map[tutorial_original] = tutorial_redirect
 
-    _possible_redirections_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS_RANDO))
+    global _possible_origins_bucket, _possible_redirections_bucket
+
+    _possible_origins_bucket = list(starmap(Transition, all_possible_transitions_rando))
+    _possible_redirections_bucket = _possible_origins_bucket.copy()
 
     if CONFIGS.LINKED_TRANSITIONS:
         # Ground rules:
         # 1. you can't make a transition from a level to itself
         # 2. any 2 levels may have a maximum of 1 connection between them (as long as it's 2-way)
 
-        _possible_origins_bucket = list(starmap(Transition, ALL_POSSIBLE_TRANSITIONS_RANDO))
+        closed_door_levels = [trans.to for trans in closed_door_exits]
+        closed_door_levels = list(dict.fromkeys(closed_door_levels))  # remove duplicates
+        random.shuffle(closed_door_levels)
 
         level_list = [
-            area for area in TRANSITION_INFOS_DICT_RANDO.values()
+            area for area in transition_infos_dict_rando.values()
             if __connections_left[area.area_id] > 0
         ]
         random.shuffle(level_list)
-        level_list.sort(key=lambda a: __connections_left[a.area_id], reverse=True)
-
-        link_list = [link_two_levels(level_list[0], level_list[1])]
-        global total_con_left
-        total_con_left = __connections_left[level_list[0].area_id]
-        total_con_left += __connections_left[level_list[1].area_id]
+        level_list.sort(
+            key=lambda a: (
+                a.area_id in closed_door_levels, __connections_left[a.area_id],
+            ), reverse=True,
+        )
 
-        index = 2
+        global __current_hub, loose_ends, total_con_left
+
+        __current_hub = closed_door_levels[0]  # this list is shuffled, so just pick the first one
+        loose_ends = [*closed_door_exits]
```

Comment:
You can create a list from an iterable, tuples are iterable. This is a bit better, even on Python 3.12 where list comprehensions are inlined
```suggestion
        loose_ends = list(closed_door_exits)
```

Reply:
Didn't know that. Sure.